src/BootConfiguration.php

```diff
@@ -0,0 +1,176 @@
+        $name = $package->getName();
+        throw new \Exception("{$name} has no valid namespace.");
+    }
+    
+    protected function getSubKeyIfItExists(array $arguments)
+    {
+        $parent = $this->rawConfig;
+        
+        foreach ($arguments as $argument){
+            if (isset($parent[$argument])) {
+                $parent = $parent[$argument];
+            }
+            else {
+                return null;
+            }
+        }
+        
+        return $parent;
+    }
+    
+    /**
+     * Tries to iterate an array element given by the arguments
+     * @param scalar $argument1,... array keys, by increasing depth
+     */
+    protected function iterateKey(...$arguments)
+    {
+        $result = $this->getSubKeyIfItExists($arguments);
+        
+        if (is_array($result)) {
+            foreach ($result as $key => $val) {
+                yield $key => $val;
+            }
+        }
+    }
+    
+    /**
+     * Returns a subkey of an array if it exists or null
+     * @param scalar $argument1,... array keys, by increasing depth
+     * @return type
+     */
+    protected function getConfig(...$arguments)
+    {
+        $result = $this->getSubKeyIfItExists($arguments);
+        return $result;
+    }
+    
+    /**
+     * internal function. Adds models to the boot configuration.
+     */
+    protected function findEntityDirectory()
+    {
+        $this->entityDirectory = null;
+        
+        $entityNamespace = $this->getConfig("bootstrap", "entityNamespace");
+        $entityNamespace = $this->rootNamespace . $entityNamespace;
+        
+        if (is_null($entityNamespace) === false) {
+            $entityDirectory = $this->composerManager->getComposer()->translateNamespaceToPath($entityNamespace);
+            
+            if (is_dir($entityDirectory) === false) {
+                throw new \Exception("{$entityDirectory}, generated from {$entityNamespace}, is not a valid directory.");
+            }
+            
+            $this->entityDirectory = $entityDirectory;
+        }
+    }
+    
+    /**
+     * Returns true if there are any models to add.
+     * @return type
+     */
+    public function hasEntityDirectory(): bool
+    {
+        return $this->entityDirectory === null ? false : true;
+    }
+    
+    /**
+     * Returns a list of fqcn for all models added by packages.
+     * @return array<string>
+     */
+    public function getEntityDirectory(): string
+    {
+        return $this->entityDirectory;
+    }
+    
+    protected function findDenerysCommands()
+    {
+        $list = $this->iterateKey("bootstrap", "daenerysCommands");
+        $this->daenerysCommands = [];
+        
+        if (is_array($list) === false) {
+            return;
+        }
+        
+        foreach ($list as $command) {
+            $this->daenerysCommands = $this->rootNamespace . $command;
+        }
+    }
+    
+    public function addDaenerysCommands(Game $game, Application $application)
```

Comment:
This function seems out of place. I would simply iterate over the boot configs in the daenerys application itself or write a helper function there instead of here.